src/main/java/io/stargate/sgv2/jsonapi/service/bridge/executor/CollectionSettings.java

```diff
@@ -106,4 +97,48 @@ public static CollectionSettings getCollectionSettings(
           null);
     }
   }
+
+  public static CollectionSettings getCollectionSettings(
+      String collectionName,
+      boolean vectorEnabled,
+      int vectorSize,
+      SimilarityFunction similarityFunction,
+      String vectorize,
+      ObjectMapper objectMapper) {
+    // parse vectorize to get vectorizeServiceName and modelName
+    if (vectorize != null && !vectorize.isBlank()) {
+      return createCollectionSettingsFromJson(
+          collectionName, vectorEnabled, vectorSize, similarityFunction, vectorize, objectMapper);
+    } else {
+      return new CollectionSettings(
+          collectionName, vectorEnabled, vectorSize, similarityFunction, null, null);
+    }
+  }
+
+  private static CollectionSettings createCollectionSettingsFromJson(
+      String collectionName,
+      boolean vectorEnabled,
+      int vectorSize,
+      SimilarityFunction function,
+      String vectorize,
+      ObjectMapper objectMapper) {
+    try {
+      JsonNode vectorizeConfig = objectMapper.readTree(vectorize);
+      String vectorizeServiceName = vectorizeConfig.path("service").textValue();
+      JsonNode optionsNode = vectorizeConfig.path("options");
+      String modelName = optionsNode.path("modelName").textValue();
+      if (vectorizeServiceName != null
+          && !vectorizeServiceName.isEmpty()
+          && modelName != null
+          && !modelName.isEmpty()) {
+        return new CollectionSettings(
+            collectionName, vectorEnabled, vectorSize, function, vectorizeServiceName, modelName);
+      } else {
+        throw new RuntimeException("Invalid json string");
+      }
+    } catch (JsonProcessingException e) {
+      // This should never happen
+      throw new RuntimeException("Invalid json string");
```

Comment:
This needs to include actual reason: add message from `e` for example, or chain it (add `e` as second argument to `RuntimeException` constructor). Otherwise there is no clue as to what is causing the exception.